### PR TITLE
[cu-23v352q][Pablo] Benchmarking and Integration to Dali

### PR DESCRIPTION
**Pull request by vimukthi-git**
- Benchmarking pablo pallet
- Integrating the pablo pallet into Dali runtime
- Minor modification to `CreationOrigin`

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-23v352q Add Pablo in Dali once everything is done](https://app.clickup.com/t/23v352q)

File: frame/pablo/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+	}
+}
+
+benchmarks! {
+  where_clause { where T::BlockNumber: From<u32>, T::Balance: From<u128>, T::AssetId: From<u128> }
+	create {
+		  let usdc: T::AssetId = 100.into();
+		  let usdt: T::AssetId = 101.into();
+		  let owner: T::AccountId = whitelisted_caller();
+		  let pair = CurrencyPair::new(usdc, usdt);
```
Comment (vivekvpandya) [inline frame/pablo/src/benchmarking.rs:57]: indentation seems wrong, also interesting that CI has not been triggered.

File: frame/pablo/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+			  initial_usdt,
+			  0.into(),
+			  false
+		  ));
+		  let user = account("user", 0, 0);
+		  assert_ok!(T::Assets::mint_into(usdt, &user, (1000_u128 * unit).into()));
+		  // swap 1000 USDC
+	 }: _(RawOrigin::Signed(user), pool_id, CurrencyPair::new(usdc, usdt), (1000_u128 * unit).into(), 0.into(), false)
+}
+
```
Comment (vivekvpandya) [inline frame/pablo/src/benchmarking.rs:274]: should we add buy/sell/swap for LBP or uniswap?

Comment (vimukthi-git) [reply frame/pablo/src/benchmarking.rs:274]: All of them are of `O(1)` complexity in terms of code, Then considering the maximum number of database operations I added stableSwap benchmarks only. For LBP I also added create/add_liquidity/remove_liquidity benchmarks as the code is some operations that are different from the other two dexes.

**Comment by seunlanlege** [comment]
You'll need to bump `impl_version` for picasso and composable too

File: runtime/dali/src/lib.rs [hunk 3]
```diff
@@ -111,7 +111,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	//   `spec_version`, and `authoring_version` are the same between Wasm and native.
 	// This value is set to 100 to notify Polkadot-JS App (https://polkadot.js.org/apps) to use
 	//   the compatible custom types.
-	spec_version: 2000,
+	spec_version: 3000,
```
Comment (seunlanlege) [inline runtime/dali/src/lib.rs:114]: should just bump impl version instead

Comment (vimukthi-git) [reply runtime/dali/src/lib.rs:114]: As we are including a new pallet, doesn't that mean we are changing the spec?

**Comment by vimukthi-git** [comment]
@seunlanlege Just for my understanding, since we only including this new pallet for dali runtime, why does the picasso and composable versions need to be updated?

**Comment by seunlanlege** [comment]
> @seunlanlege Just for my understanding, since we are only including this new pallet for dali runtime, why do picasso and composable versions need to be updated?

the reason we check if any changes have been made to the `frame/*` folder whether or not that pallet is included in the runtime, reason is because we can't figure put which runtimes are linked to each pallet.

File: runtime/dali/src/lib.rs (hunk 3 above)
Comment (seunlanlege) [reply runtime/dali/src/lib.rs:114]: well true

**Review by Dom-Roth — APPROVED** [review]
LGTM!

**Review by vivekvpandya — APPROVED** [review]

**Review by Dom-Roth — DISMISSED** [review]

**Review by Dom-Roth — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]
Simnode is failing due to new spec version but I think it should be fixed saperate.

**Comment by vimukthi-git** [comment]
Awaiting @seunlanlege to fix the simnode before merging this cc @hussein-aitlahcen

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by hussein-aitlahcen — DISMISSED** [review]

**Comment by hussein-aitlahcen** [comment]
Removed simnode as still not updated.

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Pull request merged**